### PR TITLE
Set next state

### PR DESCRIPTION
**Pull request by Kavikick**
Refactoring the process of moving one state onto the next

### CODE REVIEW

File: src/UIState/MainMenu.h [hunk 1]
```diff
@@ -6,17 +6,19 @@
 
 #pragma once
 
-#include "../TankControllerLib.h"
+// #include "../TankControllerLib.h"
```
Comment (jgfoster) [inline src/UIState/MainMenu.h]: Remove commented-out code.

**Review by jgfoster — CHANGES_REQUESTED** [review]
Remove one commented-out line of code.

File: src/UIState/NumberCollectorState.cpp [hunk 2]
```diff
@@ -10,26 +10,21 @@ void NumCollectorState::handleKey(char key) {
     numDigits++;
     handleDigit(key - '0');
     if (collectedEnoughDigits()) {
```
Comment (prestoncarman) [inline src/UIState/NumberCollectorState.cpp:12]: I thought we discussed removing `collectedEnoughDigits()`. Will that happen later?

**Review by prestoncarman — APPROVED** [review]
LGTM

File: src/UIState/NumberCollectorState.cpp (hunk 2 above)
Comment (jgfoster) [reply src/UIState/NumberCollectorState.cpp:12]: Yes, it is on its way, probably tonight!

**Review by jgfoster — APPROVED** [review]
LGTM

**Pull request merged**